components/message.php

```diff
@@ -0,0 +1,504 @@
+
+		if ( empty( $assoc_args['fields'] ) ) {
+			$assoc_args['fields'] = array_keys( $message_arr );
+		}
+
+		$formatter = $this->get_formatter( $assoc_args );
+		$formatter->display_item( $message_arr );
+	}
+
+	/**
+	 * Get a list of messages for a specific user.
+	 *
+	 * ## OPTIONS
+	 *
+	 * --user-id=<user>
+	 * : Identifier for the user. Accepts either a user_login or a numeric ID.
+	 *
+	 * [--<field>=<value>]
+	 * : One or more parameters to pass. See BP_Messages_Box_Template()
+	 *
+	 * [--fields=<fields>]
+	 * : Fields to display.
+	 *
+	 * [--count=<count>]
+	 * : How many messages to list.
+	 * ---
+	 * default: 10
+	 * ---
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - ids
+	 *   - count
+	 *   - csv
+	 *   - json
+	 *   - haml
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp message list --user-id=544 --format=count
+	 *     10
+	 *
+	 *     $ wp bp message list --user-id=user_login --count=3 --format=ids
+	 *     5454 45454 4545 465465
+	 *
+	 * @subcommand list
+	 */
+	public function _list( $_, $assoc_args ) {
+		$formatter = $this->get_formatter( $assoc_args );
+
+		$r = wp_parse_args( $assoc_args, array(
+			'box'          => 'sentbox',
+			'type'         => 'all',
+			'search'       => '',
+			'count'        => 10,
+		) );
+
+		$user = $this->get_user_id_from_identifier( $assoc_args['user-id'] );
+		if ( ! $user ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		$type = ( ! in_array( $r['type'], $this->message_types(), true ) ) ? 'all' : $r['type'];
+		$box  = ( ! in_array( $r['box'], $this->message_boxes(), true ) ) ? 'sentbox' : $r['box'];
+
+		$inbox = new BP_Messages_Box_Template( array(
+			'user_id'      => $user->ID,
+			'box'          => $box,
+			'type'         => $type,
+			'max'          => $r['count'],
+			'search_terms' => $r['search'],
+		) );
+
+		if ( ! $inbox->has_threads() ) {
+			WP_CLI::error( 'No messages found.' );
+		}
+
+		$messages = $inbox->threads[0]->messages;
+
+		if ( 'ids' === $formatter->format ) {
+			echo implode( ' ', wp_list_pluck( $messages, 'id' ) ); // WPCS: XSS ok.
+		} elseif ( 'count' === $formatter->format ) {
+			$formatter->display_items( $messages );
+		} else {
+			$formatter->display_items( $messages );
+		}
+	}
+
+	/**
+	 * Generate random messages.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--thread-id=<thread-id>]
+	 * : Thread ID to generate messages against.
+	 * ---
+	 * default: false
+	 * ---
+	 *
+	 * [--count=<number>]
+	 * : How many messages to generate.
+	 * ---
+	 * default: 20
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp message generate --thread-id=6465 --count=10
+	 *     $ wp bp message generate --count=100
+	 */
+	public function generate( $args, $assoc_args ) {
+		$notify = \WP_CLI\Utils\make_progress_bar( 'Generating messages', $assoc_args['count'] );
+
+		for ( $i = 0; $i < $assoc_args['count']; $i++ ) {
+			$this->create( array(), array(
```

Comment:
As noted above, I think random content should be generated here, not in `create()`.